daffodil-runtime1/src/main/scala/edu/illinois/ncsa/daffodil/processors/parsers/Parser.scala

```diff
@@ -173,31 +141,38 @@ class SeqCompParser(context: RuntimeData, val childParsers: Array[Parser])
 
 }
 
-class AltCompParser(context: RuntimeData, val childParsers: Seq[Parser])
-  extends ParserObject(context) {
+class AltCompParser(ctxt: RuntimeData, val childParsers: Seq[Parser])
+  extends ParserObject(ctxt) {
 
   override lazy val childProcessors = childParsers
 
   override def nom = "alt"
 
   def parse(pstate: PState): Unit = {
     var pBefore: PState.Mark = null
-
-    pstate.pushDiscriminator
-    var diagnostics: Seq[Diagnostic] = Nil
-    var i = 0
-    var parser: Parser = null
-    val limit = childParsers.length
-    var returnFlag = false
-    while (!returnFlag && i < limit) {
-      parser = childParsers(i)
-      i += 1
-      log(LogLevel.Debug, "Trying choice alternative: %s", parser)
-      pBefore = pstate.mark("AltCompParser1")
-      try {
-        parser.parse1(pstate)
-      } catch {
-        case d: SchemaDefinitionDiagnosticBase => {
+    try {
+      pstate.pushDiscriminator
+      var diagnostics: Seq[Diagnostic] = Nil
+      var i = 0
+      var parser: Parser = null
+      val limit = childParsers.length
+      var returnFlag = false
+      while (!returnFlag && i < limit) {
+        parser = childParsers(i)
+        i += 1
+        log(LogLevel.Debug, "Trying choice alternative: %s", parser)
+        pBefore = pstate.mark("AltCompParser1")
+        try {
+          pstate.parse1(parser)
```

Comment:
Fix up per above.

Reply:
Not sure what this change is? Maybe something failed with rebasing with the commit to clean up Mark Pools?

Reply:
Yeah, I made a mistake in merging this file, ran sbt test in a different directory, etc. Net result, this file doesn't even compile as it appears here......